openfisca_core/parameters.py

```diff
@@ -263,20 +263,24 @@ def validate(self, data):
             value = data['value']
         except KeyError:
             raise ParameterParsingError(
-                "Missing 'value' property for {}".format(self.name).encode('utf-8'),
+                u"Missing 'value' property for {}".format(self.name),
                 self.file_path
                 )
         if type(value) not in self._allowed_value_data_types:
             raise ParameterParsingError(
-                "Invalid value in {} : {}".format(self.name, value).encode('utf-8'),
+                u"Invalid value in {} : {}".format(self.name, value),
                 self.file_path
                 )
 
     def __eq__(self, other):
         return (self.name == other.name) and (self.instant_str == other.instant_str) and (self.value == other.value)
 
     def __repr__(self):
-        return "ParameterAtInstant({})".format({self.instant_str: self.value}).encode('utf-8')
+        result = "ParameterAtInstant({})".format({self.instant_str: self.value})
+        # repr output must be encoded in Python 2, but not in Python 3
+        if sys.version_info < (3, 0):
```

Comment:
Coult this be a `to_utf8` function ?It seem to happen several times in the code

Reply:
In both Python 2 and 3, `__repr__` must return a `str`. But `str` has a different meaning in Python 2 and 3 😒: it is a `byte` in Python 2, but a `unicode` in Python 3.

We only want to encode the string in `UTF-8` in Python 2, as in Python 3 we want to return a unicode. So if if we extracted the function, that would be `to_utf8_if_python2`, which doesn't seem great.

If all this encoding business is not clear for you, as it was for me a few days ago,  I'd recommend [this article[FR]](http://sametmax.com/lencoding-en-python-une-bonne-fois-pour-toute/). I can't pretend I understand everything now, but it's more clear 🙂.